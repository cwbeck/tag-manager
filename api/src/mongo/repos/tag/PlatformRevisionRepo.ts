import Repo from '../../abstractions/Repo';
import { injectable } from 'inversify';
import PlatformRevision from '../../models/tag/PlatformRevision';

@injectable()
export default class PlatformRevisionRepo extends Repo<PlatformRevision> {
    protected readonly auditEnabled = true;
}
