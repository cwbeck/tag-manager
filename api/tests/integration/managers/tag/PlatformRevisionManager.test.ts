import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import PlatformRevisionRepo from '../../../../src/mongo/repos/tag/PlatformRevisionRepo';
import Mock = jest.Mock;

describe('PlatformRevisionManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a single platform revision', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockOrgUserWithViewAccessOnPlatformRevision();

        const mockSutModel = TestUtils.buildMockPlatformRevision();
        const mockSutRepo = {} as PlatformRevisionRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getPlatformRevision(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
