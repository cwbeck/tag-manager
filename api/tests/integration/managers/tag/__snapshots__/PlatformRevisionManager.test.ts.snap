// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformRevisionManager fetches a single platform revision 1`] = `
Object {
  "data": Object {
    "getPlatformRevision": Object {
      "id": "5edf4c48e9615d9a1c66cbbb",
      "name": "test platform revision",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
